modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -1340,19 +1445,34 @@ GrainTracker::boundingRegionDistance(std::vector<MeshTools::BoundingBox> & bboxe
   return min_distance;
 }
 
+unsigned int
+GrainTracker::getNextUniqueID()
+{
+  /**
+   * Get the next unique grain ID but make sure to respect
+   * reserve ids.
+   */
+  while (++_max_curr_grain_id <= _reserve_grain_first_idx + _n_reserve_ops)
```

Comment:
Not quite! Your statement is not equivalent because _max_curr_grain_id might be greater than the lhs. However this would work too:
_max_curr_grain_id = std::max(_max_curr_grain+id + 1, _reserve_grain_first_idx + _n_reserve_ops);